crates/uv/src/commands/project/sync.rs

```diff
@@ -1294,6 +1306,77 @@ impl SyncReport {
     }
 }
 
+/// A summary of a single package change performed during sync.
+#[derive(Serialize, Debug, Clone)]
+struct PackageChangeReport {
```

Comment:
Have you thought more about what other information we'll want to include here? I don't think we need to expand on it in this pull request but we should at least open an issue for discussion.